public/app/plugins/datasource/prometheus/language_provider.ts

```diff
@@ -116,6 +121,11 @@ export default class PromQlLanguageProvider extends LanguageProvider {
     this.lookupsDisabled = this.metrics.length > this.lookupMetricsThreshold;
     this.metricsMetadata = await this.request('/api/v1/metadata', {});
     this.processHistogramMetrics(this.metrics);
+
+    if (!this.lookupsDisabled) {
+      truncateExcessivelyLongMetricNames(this.metrics, this.metricsMetadata);
```

Comment:
It would be great to have some explanation comment why we are doing this and why it is ok to do.